core/src/main/scala/org/apache/spark/sql/internal/SnappySessionCatalog.scala

```diff
@@ -752,7 +753,7 @@ class SnappySessionCatalog(val externalCatalog: SnappyExternalCatalog,
   private def removeFromFuncJars(funcDefinition: CatalogFunction,
       qualifiedName: FunctionIdentifier): Unit = {
     funcDefinition.resources.foreach { r =>
-      ContextJarUtils.deleteFile(funcDefinition.identifier.toString(), r.uri)
+      ContextJarUtils.deleteFile(funcDefinition.identifier.toString(), r.uri, isEmbeddedMode())
```

Comment:
Dropping from smart connector will not clean the lead file neither the embedded server's local file. We need to clean both smart connector cluster as well as embedded cluster. You can call a stored procedure or something for the smart connector. Please check GET_DEPLOYED_JARS procedure. We can have a similar procedure for this cleanup.